src/install/install.zig

```diff
@@ -487,30 +488,41 @@ const Task = struct {
 
     /// An ID that lets us register a callback without keeping the same pointer around
     pub const Id = struct {
-        pub fn forNPMPackage(_: Task.Tag, package_name: string, package_version: Semver.Version) u64 {
+        pub fn forNPMPackage(package_name: string, package_version: Semver.Version) u64 {
             var hasher = std.hash.Wyhash.init(0);
             hasher.update(package_name);
             hasher.update("@");
             hasher.update(std.mem.asBytes(&package_version));
-            return @as(u64, @truncate(u62, hasher.final())) | @as(u64, 1 << 63);
+            return @as(u64, 0 << 61) | @as(u64, @truncate(u61, hasher.final()));
         }
 
         pub fn forBinLink(package_id: PackageID) u64 {
             const hash = std.hash.Wyhash.hash(0, std.mem.asBytes(&package_id));
-            return @as(u64, @truncate(u62, hash)) | @as(u64, 1 << 62) | @as(u64, 1 << 63);
+            return @as(u64, 1 << 61) | @as(u64, @truncate(u61, hash));
         }
 
-        pub fn forManifest(
-            _: Task.Tag,
-            name: string,
-        ) u64 {
-            return @as(u64, @truncate(u62, std.hash.Wyhash.hash(0, name)));
+        pub fn forManifest(name: string) u64 {
+            return @as(u64, 2 << 61) | @as(u64, @truncate(u61, std.hash.Wyhash.hash(0, name)));
         }
 
         pub fn forTarball(url: string) u64 {
             var hasher = std.hash.Wyhash.init(0);
             hasher.update(url);
-            return @as(u64, @truncate(u62, hasher.final())) | @as(u64, 1 << 62);
+            return @as(u64, 3 << 61) | @as(u64, @truncate(u61, hasher.final()));
+        }
+
+        pub fn forGitClone(url: string) u64 {
+            var hasher = std.hash.Wyhash.init(0);
+            hasher.update(url);
+            return @as(u64, 4 << 61) | @as(u64, @truncate(u61, hasher.final()));
+        }
+
+        pub fn forGitCheckout(url: string, resolved: string) u64 {
+            var hasher = std.hash.Wyhash.init(0);
+            hasher.update(url);
+            hasher.update("@");
+            hasher.update(resolved);
+            return @as(u64, 5 << 61) | @as(u64, @truncate(u61, hasher.final()));
```

Comment:
5 + 61 == 66?

I think we need to make the right hand side u59

Reply:
`64 - 61 = 3` bits, and `2 ** 3 = 8` so we have `0 .. 7` to play with here? :thinking:

Reply:
ah yeah